### PR TITLE
refactor(demo): Use React Hooks

### PR DESCRIPTION
**Pull request by mttrbrts**
image

### Changes
- Rewrite demo app using React Functional components and Hooks
- Use Monaco editors for Concerto and JSON
- Better handling of errors in JSON editor
- Adds Loader during async loading
- Removes need for form ref in client app

Preview at: https://deploy-preview-36--concerto-ui.netlify.com/

### CODE REVIEW

**Review by jeromesimeon — APPROVED** [review]
This looks great!
PS: I'm not a react hook expert.

File: packages/concerto-ui-react/src/concertoForm.js [hunk 1]
```diff
@@ -114,12 +117,23 @@ class ConcertoForm extends Component {
...
   }
 
-  static getDerivedStateFromProps(props, state){
-    return { value: props.json, warning: null};
+  static getDerivedStateFromProps(props){
+    let shadowValue;
+    if(typeof props.json === 'string'){
+      try{
+        shadowValue = (JSON.parse(props.json));
+      } catch {}
```
Comment (DianaLease) [inline packages/concerto-ui-react/src/concertoForm.js:132]: should `warning` be set in this `catch` block?

**Review by DianaLease — COMMENTED** [review]
Looks good! Left one comment.

File: packages/concerto-ui-react/src/concertoForm.js (hunk 1 above)
Comment (mttrbrts) [reply packages/concerto-ui-react/src/concertoForm.js:132]: Thanks, `warning` isn't actually used anymore, so I've removed it elsewhere too

**Review by DianaLease — APPROVED** [review]

**Pull request merged**